packages/cubit/lib/src/cubit.dart

```diff
@@ -47,14 +47,18 @@ abstract class Cubit<T> extends Stream<T> {
 
   /// Updates the [state] of the `cubit` to the provided [state].
   /// [emit] does nothing if the `cubit` has been closed.
+  ///
+  /// [track] can optionally be provided to determine whether the
+  /// state change should be tracked (reported to `onTransition`).
+  /// Default is `true`.
   @protected
-  void emit(T state) async {
+  void emit(T state, {bool track}) async {
     if (state == _state || _controller.isClosed) return;
-    onTransition(Transition(
-      currentState: _state,
-      event: null,
-      nextState: state,
-    ));
+    if (track != false) {
```

Comment:
```suggestion
    if (track == true) {
```

Reply:
NNBD coming soon. :-) @felangel we might port this library to NNBD now. It should be backward compatible.

Reply:
Yeah good point! I’ll give it a shot haha 😂